### ISSUE TITLE
Exception on Android: com/fiskaly/kassensichv/sma/Sma

### ISSUE THREAD

**Issue by cjaentsch** [fiskaly/fiskaly-kassensichv-client-java#14]
When trying to use the SMA on Android I get the following error...
I use the latest master commit (as of today). I put the client.android.jar and client.general.jar to our app and give an instance of AndroidSMA to the ClientFactory.

Failed resolution of: Lcom/fiskaly/kassensichv/sma/Sma;: java.lang.NoClassDefFoundError: Failed resolution of: Lcom/fiskaly/kassensichv/sma/Sma;↵	at com.fiskaly.kassensichv.sma.AndroidSMA.invoke(AndroidSMA.java:6)↵	at com.fiskaly.kassensichv.client.interceptors.TransactionInterceptor.signTransaction(TransactionInterceptor.java:48)↵	at com.fiskaly.kassensichv.client.interceptors.TransactionInterceptor.intercept(TransactionInterceptor.java:129)↵	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)↵	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)↵	at com.fiskaly.kassensichv.client.interceptors.AuthenticationInterceptor.intercept(AuthenticationInterceptor.java:49)↵	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)↵	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)↵	at com.fiskaly.kassensichv.client.interceptors.HeaderInterceptor.intercept(HeaderInterceptor.java:23)↵	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)↵	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)↵	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.kt:184)↵	at okhttp3.RealCall.execute(RealCall.kt:66)

**Comment by cjaentsch** [comment]
Seems to be an issue with our obfuscation process. We'll check again and keep you updated.

**Comment by cjaentsch** [comment]
OK, so we have investigated further (together with the new fiskaly client jar build released today)...

When using the obfuscator, we still get the error mentioned above on runtime. When not using the obfuscator (the main jar of our application), we get an error during the build process of the Cordova app:

D8: Program type already present: com.fiskaly.kassensichv.client.interceptors.AuthenticationInterceptor
D8: Program type already present: com.fiskaly.kassensichv.client.authentication.Authenticator
D8: Program type already present: com.fiskaly.kassensichv.client.interceptors.HeaderInterceptor
D8: Program type already present: com.fiskaly.kassensichv.client.ClientFactory
com.android.builder.dexing.DexArchiveBuilderException: Failed to proces

I really don't have a clue what to do next. I tried with less included libraries and so on. Maybe the obfuscator generally is correct but has a problem because we have a "com.fiskaly.kassensichv.sma.SMA" and a "com.fiskaly.kassensichv.sma.Sma" (so same letters but one with capital letters).
